e2e/test/transactions.js

```diff
@@ -27,14 +28,34 @@ describe('transactions', () => {
     assert(toBN(balance).isZero(), 'Account should not have tokens yet')
 
     // send transaction to home chain
-    await homeWeb3.eth.sendTransaction({
+    const depositTx = await homeWeb3.eth.sendTransaction({
       from: user.address,
       to: HOME_BRIDGE_ADDRESS,
       gasPrice: '1',
-      gasLimit: '50000',
+      gas: '50000',
       value: '1000000000000000000'
     })
 
+    // Send a Tx to generate a new block
+    await generateNewBlockWithTx(homeWeb3, user.address)
+
+    // wait to send a Tx to generate a new block
+    await promiseRetry(
+      async retry => {
+        const lastBlockNumber = await homeWeb3.eth.getBlockNumber()
+        if (lastBlockNumber === depositTx.blockNumber + 2) {
```

Comment:
I would use `>=` here. It's very unlikely for this to be necessary, but it won't hurt and it's a little more defensive.